app/qml/FormsStackManager.qml

```diff
@@ -120,6 +121,32 @@ Item {
     formsStack.visible = false
   }
 
+  function openLinkedFeature( linkedFeature ) {
+    let props = {
+      featureLayerPair: linkedFeature,
+      formState: "ReadOnly",
+      panelState: "form"
+    }
+
+    let latest = formsStack.push( formComponent, props )
+    root.activeFormIndex = latest.StackView.index
+  }
+
+  function createLinkedFeature( parentController, relation ) {
+    let newFeaturePair = digitizingController.featureWithoutGeometry( relation.referencingLayer )
```

Comment:
it would be nice to have forms decoupled from digitizingController - and the featureWithoutGeometry() function could be e.g. in InputUtils or a similar place... but no strong feelings about that

Reply:
Hm, I am not sure, it is not a static function. It uses variables manager and evaluates the variables in feature's context. 

See https://github.com/lutraconsulting/input/pull/1531/files#diff-a414e2e410de55a5bc2ba342f687a3a7d92d664079d2f7e977217403c53c46beR220-R227

If it would be simple `QgsVectorUtils::createFeature` then I would go for it, but with variables manager I am not really sure. What do you think? We would need to add the variables manager then to inpututils and somehow change it to be static

Reply:
I see -- yeah let's keep it as it is for now... and maybe in the future we can have some class that has the project context in it, including variables manager, decoupled from digitizing controller...